sdk/include/opentelemetry/sdk/trace/sampler.h

```diff
@@ -0,0 +1,56 @@
+#pragma once
+
+#include "opentelemetry/version.h"
+#include "opentelemetry/common/attribute_value.h"
+
+#include <memory>
+#include <map>
+#include <string>
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace trace
+{
+class Sampler
+{
+public:
+  /**
+   * Initialize a new tracer.
+   * @param processor The span processor for this tracer. This must not be a
+   * nullptr.
+   */
+  virtual Decision shouldSample();
+
+  /**
+   * Returns the description of this {@code Sampler}. This may be displayed on debug pages or in the
+   * logs.
+   *
+   * <p>Example: "ProbabilitySampler{0.000100}"
+   *
+   * @return the description of this {@code Sampler}.
+   */
+  virtual std::string getDescription();
+};
+
+class Decision
```

Comment:
This class should be called `SamplingResult`, according to the [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/sdk.md#shouldsample).

Reply:
Probably @ziqizh was following OpenTelemetry Java? I would say take a look at Java (just to get some idea) but meanwhile stick with the spec. 😄

Reply:
Thank you for pointing this out. Yes, I copy-pasted some definitions from Java, and I will double-check all these classes tomorrow!